bitnami/solr/templates/statefulset.yaml

```diff
@@ -0,0 +1,237 @@
+          {{- end }}
+          {{- if .Values.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
+          {{- else }}
+          command: 
+            - bash
+            - -ec
+            - |
+              {{- if .Values.cloudBootstrap }}
+              NODE_ID="${MY_POD_NAME#"{{ include "solr.fullname" . }}-"}"
+              if [[ "$NODE_ID" -eq "0" ]]; then
+                export SOLR_CLOUD_BOOTSTRAP=yes
+              fi
+              {{- end }}
+              # Use hostname instead of IP to register in Zookeeper
+              export SOLR_HOST="${MY_POD_NAME}.{{ include "solr.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}"
+              /opt/bitnami/scripts/solr/entrypoint.sh /opt/bitnami/scripts/solr/run.sh
+          {{- end }}
+          {{- if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: MY_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: SOLR_ENABLE_CLOUD_MODE
+              value: {{ ternary "yes" "no" .Values.cloudEnabled | quote }}
+            - name: SOLR_NUMBER_OF_NODES      
+              value: {{ .Values.replicaCount | quote }}
+            - name: SOLR_PORT_NUMBER      
+              value: {{ .Values.containerPort | quote }}
+            - name: SOLR_SERVER_DIRECTORY
+              value: {{ .Values.serverDirectory | quote }}
+            {{- if .Values.cloudEnabled }}
+            - name: SOLR_COLLECTION
+              value: {{ .Values.collection | quote }}
+            {{- else }}
+            - name: SOLR_CORE
+              value: {{ .Values.coreName | quote }}
+            {{- end }}
+            - name: SOLR_COLLECTION_SHARDS
+              value: {{ .Values.collectionShards | quote }}
+            - name: SOLR_COLLECTION_REPLICAS
+              value: {{ .Values.collectionReplicas | quote }}
+            {{- if .Values.heap }}
+            - name: SOLR_HEAP
+              value: {{ .Values.heap | quote }}
+            {{- end }}
+            {{- if .Values.javaMem }}
+            - name: SOLR_JAVA_MEM
+              value: {{ .Values.javaMem | quote }}
+            {{- end }}
+            {{- if .Values.authentication.enabled }}
+            - name: SOLR_ENABLE_AUTHENTICATION
+              value: "yes"
+            - name: SOLR_ADMIN_USERNAME
+              value: {{ .Values.authentication.adminUsername | quote }}
+            - name: SOLR_ADMIN_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "solr.secretName" . }}
+                  key: solr-password
+            - name: SOLR_ZK_HOSTS
+              value: "{{ include "solr.zookeeper.host" . }}:{{ .Values.zookeeper.port }}"
```

Comment:
In the case of multiple zookeepers (for instance, using the external zookeeper, which uses an array), don't you have to set the port in each zk instance?

Maybe it makes sense to completely move it to a helper.